src/test/java/io/reactivex/processors/MulticastProcessorTest.java

```diff
@@ -0,0 +1,793 @@
+
+import org.junit.Test;
+import org.reactivestreams.Subscription;
+
+import io.reactivex.*;
+import io.reactivex.exceptions.*;
+import io.reactivex.functions.Function;
+import io.reactivex.internal.subscriptions.BooleanSubscription;
+import io.reactivex.plugins.RxJavaPlugins;
+import io.reactivex.subscribers.TestSubscriber;
+
+public class MulticastProcessorTest {
+
+    @Test
+    public void complete() {
+        MulticastProcessor<Integer> mp = MulticastProcessor.create();
+        mp.start();
+
+        assertFalse(mp.hasSubscribers());
+        assertFalse(mp.hasComplete());
+        assertFalse(mp.hasThrowable());
+        assertNull(mp.getThrowable());
+
+        TestSubscriber<Integer> ts = mp.test();
+
+        assertTrue(mp.hasSubscribers());
+        assertFalse(mp.hasComplete());
+        assertFalse(mp.hasThrowable());
+        assertNull(mp.getThrowable());
+
+        mp.onNext(1);
+        mp.onComplete();
+
+        ts.assertResult(1);
+
+        assertFalse(mp.hasSubscribers());
+        assertTrue(mp.hasComplete());
+        assertFalse(mp.hasThrowable());
+        assertNull(mp.getThrowable());
+
+        mp.test().assertResult();
+    }
+
+
+    @Test
+    public void error() {
+        MulticastProcessor<Integer> mp = MulticastProcessor.create();
+        mp.start();
+
+        assertFalse(mp.hasSubscribers());
+        assertFalse(mp.hasComplete());
+        assertFalse(mp.hasThrowable());
+        assertNull(mp.getThrowable());
+
+        TestSubscriber<Integer> ts = mp.test();
+
+        assertTrue(mp.hasSubscribers());
+        assertFalse(mp.hasComplete());
+        assertFalse(mp.hasThrowable());
+        assertNull(mp.getThrowable());
+
+        mp.onNext(1);
+        mp.onError(new IOException());
+
+        ts.assertFailure(IOException.class, 1);
+
+        assertFalse(mp.hasSubscribers());
+        assertFalse(mp.hasComplete());
+        assertTrue(mp.hasThrowable());
+        assertNotNull(mp.getThrowable());
+        assertTrue("" + mp.getThrowable(), mp.getThrowable() instanceof IOException);
+
+        mp.test().assertFailure(IOException.class);
+    }
+
+    @Test
+    public void overflow() {
+        MulticastProcessor<Integer> mp = MulticastProcessor.create(1);
+        mp.start();
+
+        TestSubscriber<Integer> ts = mp.test(0);
+
+        assertTrue(mp.offer(1));
+        assertFalse(mp.offer(2));
+
+        mp.onNext(3);
+
+        ts.assertEmpty();
+
+        ts.request(1);
+
+        ts.assertFailure(MissingBackpressureException.class, 1);
+
+        mp.test().assertFailure(MissingBackpressureException.class);
+    }
+
+    @Test
+    public void backpressure() {
+        MulticastProcessor<Integer> mp = MulticastProcessor.create(16, false);
+        mp.start();
+
+        for (int i = 0; i < 10; i++) {
+            mp.onNext(i);
+        }
+        mp.onComplete();
+
+        mp.test(0)
+        .assertEmpty()
+        .requestMore(1)
+        .assertValues(0)
+        .assertNotComplete()
```

Comment:
Could you just use `assertValuesOnly` and then save the `assertNotComplete` call?